### PR TITLE
Upgrade to gophercloud 2.0.0

### PR DESCRIPTION
**Pull request by SuperSandro2000**

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
checks will be fixed when the action is available on master with the updated go version.

**Comment by kayrus** [comment]
@SuperSandro2000 I'll take care of the interconnections package.

File: resources/v1/clusters/testing/requests_test.go [hunk 1]
```diff
@@ -136,7 +137,7 @@ func TestGetFilteredCluster(t *testing.T) {
 						Capacity:      &capacity,
 						DomainsQuota:  p2ui64(4),
 						Usage:         2,
-						Subcapacities: json.RawMessage(`[{"cores":200,"hypervisor":"cluster-1"},{"cores":800,"hypervisor":"cluster-2"}]`),
+						Subcapacities: json.RawMessage(`[{"cores":http.StatusOK,"hypervisor":"cluster-1"},{"cores":800,"hypervisor":"cluster-2"}]`),
```
Comment (majewsky) [inline resources/v1/clusters/testing/requests_test.go:140]: This is not valid JSON. Run-away regex replacement? :)

**Review by majewsky — CHANGES_REQUESTED** [review]
Rest looks good.

**Review by majewsky — APPROVED** [review]

**Pull request merged**